### DISCUSSION TITLE
Database function returns empty response with RPC call, but works everywhere else? Even with RLS disabled?

### DISCUSSION THREAD

**Discussion by stanifert** [supabase/supabase#21917] [Questions] [answered]
I have a database function with return type of `jsonb` and security set as `invoker`. The function queries a table and returns matching records as json. The table is of schema `public` and has 1 active RLS policy which enables read access for all users to the table.

Using the Swift client (2.1.0), I can successfully query the table directly and return data. However, when I try to call my database function via RPC, I get an empty response.

What I have tried:
- I tried disabling RLS altogether from the table, no change
- I tried changed database function to security `definer` instead, no change
- I can log out the generic `PostgrestResponse` object (from `.execute()`) and see that it has a 200 OK response, but the data is not there
- I can call the same database function successfully via REST api using my public key

I suspect I'm missing something RLS/roles/security related but am stumped since everything seems to be public/accessible. Any help would be appreciated! 

My function:
```
declare
	item_array jsonb[];
begin
	select array_agg(row_to_json(item)) table_name
	into item_array
	from (
		select *
		from table_name
		where created_at::timestamp < now()::timestamp
		order by created_at asc
	) item;

	return (select jsonb_build_object(
		'results', item_array
	));
end;
```

The below client code gives the error: `The data couldn’t be read because it is missing`
```
try await supabase.database
    .rpc("my_function")
    .execute()
    .value
```
but this works without issue:
```
try await supabase.database
    .from("table_name")
    .select()
    .execute()
```

**Comment by GaryAustin1** [comment]
Add `raise log 'Function running';` and make sure this function is being called by looking for the message in the Postgres logs of the dashboard.

**Reply by stanifert** [reply]
Thank you for that! Confirmed that the function is being called via `.rpc(...)` on the swift client, can see both the Postman call and client call in Postgres logs in dashboard, but still no data w/ RPC call unfortunately

**Comment by stanifert** [comment, marked as answer]
Seems that I was just misusing the Swift client. I was following the example [here](https://supabase.com/docs/reference/swift/using-filters) under "Call a Postgres function" like so:
```
try await supabase.database
    .rpc("my_function")
    .execute()
    .value
```

For some reason I was expecting the `.value` to parse the data received into my codable object, but that seems to not the case. The below code works as expected:
```
let data = try await supabase.database
    .rpc("my_function")
    .execute()
    .data
return try decoder.decode(MyCodableObject.self, from: data)
```

@GaryAustin1 thank you for the help!
